packages/readmodel/src/agreement/splitters.ts

```diff
@@ -0,0 +1,235 @@
+import {
+  Agreement,
+  AgreementDocument,
+  AgreementId,
+  AgreementStamp,
+  AgreementStampKind,
+  AgreementStamps,
+  attributeKind,
+  dateToString,
+} from "pagopa-interop-models";
+import {
+  AgreementSQL,
+  AgreementStampSQL,
+  AgreementAttributeSQL,
+  AgreementConsumerDocumentSQL,
+  AgreementContractSQL,
+  AgreementItemsSQL,
+} from "pagopa-interop-readmodel-models";
+
+export const splitAgreementIntoObjectsSQL = (
+  {
+    id,
+    eserviceId,
+    descriptorId,
+    producerId,
+    consumerId,
+    state,
+    verifiedAttributes,
+    certifiedAttributes,
+    declaredAttributes,
+    suspendedByConsumer,
+    suspendedByProducer,
+    suspendedByPlatform,
+    consumerDocuments,
+    createdAt,
+    updatedAt,
+    consumerNotes,
+    contract,
+    stamps,
+    rejectionReason,
+    suspendedAt,
+    ...rest
+  }: Agreement,
+  metadataVersion: number
+): AgreementItemsSQL => {
+  void (rest satisfies Record<string, never>);
+  const agreementSQL: AgreementSQL = {
+    id,
+    metadataVersion,
+    eserviceId,
+    descriptorId,
+    producerId,
+    consumerId,
+    state,
+    suspendedByConsumer: suspendedByConsumer ?? null, // "??" because "false" should not become null
+    suspendedByProducer: suspendedByProducer ?? null,
+    suspendedByPlatform: suspendedByPlatform ?? null,
+    createdAt: dateToString(createdAt),
+    updatedAt: dateToString(updatedAt),
+    consumerNotes: consumerNotes ?? null, // "??" because empty strings should not become null
+    rejectionReason: rejectionReason ?? null,
+    suspendedAt: dateToString(suspendedAt),
+  };
+
+  const stampsSQL: AgreementStampSQL[] = [];
+
+  const makeStampSQL = (
+    { who, delegationId, when, ...stampRest }: AgreementStamp,
+    agreementId: AgreementId,
+    metadataVersion: number,
+    kind: AgreementStampKind
+  ): AgreementStampSQL => {
+    void (stampRest satisfies Record<string, never>);
+
+    return {
+      agreementId,
+      metadataVersion,
+      kind,
+      who,
+      when: dateToString(when),
+      delegationId: delegationId || null,
+    };
+  };
+
+  // TODO: improve?
+  // eslint-disable-next-line functional/no-let
+  let key: keyof AgreementStamps;
+
+  // eslint-disable-next-line guard-for-in
+  for (key in stamps) {
+    const stamp = stamps[key];
+    if (stamp) {
+      // eslint-disable-next-line functional/immutable-data
+      stampsSQL.push(
+        makeStampSQL(stamp, id, metadataVersion, AgreementStampKind.enum[key])
+      );
+    }
+  }
```

Comment:
```suggestion
  const stampsSQL: AgreementStampSQL[] = Object.entries(stamps)
    .filter((entry): entry is [AgreementStampKind, AgreementStamp] => {
      const [, stamp] = entry;
      return stamp !== undefined;
    })
    .map(([key, stamp]) => makeStampSQL(stamp, id, metadataVersion, key));
```
I didn't test it but we could something like this